src/org/traccar/database/DeviceManager.java

```diff
@@ -52,6 +55,13 @@
 
     private final Map<Long, Position> positions = new ConcurrentHashMap<>();
 
+    private final Map<Long, Boolean> motionStates = new ConcurrentHashMap<>();
+    private final Map<Long, Position> motionStatePositions = new ConcurrentHashMap<>();
```

Comment:
Don't we already store positions above? or is it a first position? In that case do we really need to store the whole position or just the time would be enough?